### ISSUE TITLE
Getting state_changes fails with "you don't have permission for idle" on password protected MPD

### ISSUE THREAD

**Issue by dragonwit** [elomatreb/mpd_client#10]
I'm trying to connect to a MPD with everything password protected. I provide correct password with "password" command. Executing commands such as "next" works, but when trying to get state change with state_changes.next() I get this error:

Error: ErrorMessage(Error { code: 4, command_index: 0, current_command: Some("idle"), message: "you don't have permission for \"idle\"" })

Perhaps this has something to do with mpd_client sending "idle" command before it is even possible to send password to MPD. Or maybe I'm missing something? (I'm new with rust)

**Comment by elomatreb** [comment]
Hi, this is indeed caused by the server not accepting the `idle` command before the password, so a bug in this library. I didn't test with a password-protected server properly it seems.

I'll publish a version with a fix together with some other changes I was planning to make.

**Comment by elomatreb** [comment]
I implemented a new password system in 05aac5bfcd5ab9f0ef1991ec3284d9ca2f423c58. Passwords are now specified during the initial connect, before any automatic commands are sent, and you will immediately get an error if the supplied password is not accepted.

Connecting would something like this:

```rs
let mut connection = TcpStream::connect("localhost:6600").await.unwrap();
let (client, state_changes) = Client::connect_with_password(connection, "your_password_here").await.unwrap();
```

I'll publish new releases containing this change shortly, assuming you don't see any major problems with this solution.

**Comment by dragonwit** [comment]
Seems to work great.

**Comment by elomatreb** [comment]
I just published v0.7.0 of this crate, which includes this change. Sorry for the delay, I was busy with exams at work.
